sn_client/src/audit/spend_dag.rs

```diff
@@ -10,12 +10,35 @@ use petgraph::dot::Dot;
 use petgraph::graph::{DiGraph, NodeIndex};
 use petgraph::visit::EdgeRef;
 use serde::{Deserialize, Serialize};
-use sn_transfers::{is_genesis_spend, CashNoteRedemption, NanoTokens, SignedSpend, SpendAddress};
-use std::collections::{BTreeMap, BTreeSet};
-use std::path::Path;
+use sn_transfers::{
+    is_genesis_spend, CashNoteRedemption, NanoTokens, SignedSpend, SpendAddress,
+    CASHNOTE_PURPOSE_OF_GENESIS, CASHNOTE_PURPOSE_OF_NETWORK_ROYALTIES,
+};
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    fmt,
+    path::Path,
+};
 
 use super::dag_error::{DagError, SpendFault};
 
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+pub struct SpendDagAddress {
+    address: SpendAddress,
+    purpose: String,
+}
+
+impl std::fmt::Debug for SpendDagAddress {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "address: {}\npurpose: {}",
+            &self.address.to_hex()[0..6],
+            self.purpose
+        )
+    }
+}
```

Comment:
```suggestion
#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub struct SpendDagAddress {
    address: SpendAddress,
    purpose: String,
}
```

Reply:
the default Debug doesn't have the `line break`, which gives a better output in the diagraph.
hence it has to be individually implemented.